fieldpath/element.go

```diff
@@ -47,6 +47,16 @@ type PathElement struct {
 	Index *int
 }
 
+// NewValuePathElement creates a PathElement from the provided value.Value
```

Comment:
Can you remove these two functions if we don't need them? I suspect we already have something like this in `MakePath`